microservices-orders/start-ccloud.sh

```diff
@@ -0,0 +1,100 @@
+
+printf "\n====== Creating demo topics\n"
+./scripts/create-topics-ccloud.sh ./topics.txt
+
+printf "\n====== Starting local services in Docker\n"
+docker-compose -f docker-compose-ccloud.yml up -d --build 
+
+printf "\n====== Giving services $WARMUP_TIME seconds to startup\n"
+sleep $WARMUP_TIME 
+
+printf "\n====== Configuring Elasticsearch mappings\n"
+./dashboard/set_elasticsearch_mapping.sh
+
+printf "\n====== Submitting connectors\n\n"
+printf "====== Submitting Kafka Connector to source customers from sqlite3 database and produce to topic 'customers'\n"
+INPUT_FILE=./connectors/connector_jdbc_customers_template.config 
+OUTPUT_FILE=./connectors/rendered-connectors/connector_jdbc_customers.config 
+SQLITE_DB_PATH=/opt/docker/db/data/microservices.db
+source ./scripts/render-connector-config.sh 
+curl -s -S -XPOST -H Accept:application/json -H Content-Type:application/json http://localhost:8083/connectors/ -d @$OUTPUT_FILE
+
+printf "\n\n====== Submitting Kafka Connector to sink records from 'orders' topic to Elasticsearch\n"
+INPUT_FILE=./connectors/connector_elasticsearch_template.config
+OUTPUT_FILE=./connectors/rendered-connectors/connector_elasticsearch.config 
+ELASTICSEARCH_URL=http://elasticsearch:9200
+source ./scripts/render-connector-config.sh
+curl -s -S -XPOST -H Accept:application/json -H Content-Type:application/json http://localhost:8083/connectors/ -d @$OUTPUT_FILE
+
+printf "\n====== Validting and setting up ksqlDB App\n"
+
+MAX_WAIT_KSQLDB=$MAX_WAIT
+printf "\n====== Waiting up to $MAX_WAIT_KSQLDB for ksqlDB to be ready\n"
+retry $MAX_WAIT ccloud::validate_ccloud_ksqldb_endpoint_ready $KSQLDB_ENDPOINT || exit 1
+
+printf "====== Creating ksqlDB ACLs\n"
+ccloud kafka acl create --allow --service-account $KSQLDB_SERVICE_ACCOUNT_ID --operation READ --topic 'orders'
+printf "\n"
+ccloud kafka acl create --allow --service-account $KSQLDB_SERVICE_ACCOUNT_ID --operation READ --topic 'customers'
+printf "\n"
+
+printf "\n====== Creating ksqlDB STREAMS and TABLES\n"
+KQLDB_QUERY_PROPERTIES='"ksql.streams.auto.offset.reset":"earliest","ksql.streams.cache.max.bytes.buffering":"0"'
+while read ksqlCmd; do
+  printf "\n$ksqlCmd\n"
+  response=$(curl -X POST $KSQLDB_ENDPOINT/ksql \
+       -H "Content-Type: application/vnd.ksql.v1+json; charset=utf-8" \
+       -u $KSQLDB_BASIC_AUTH_USER_INFO \
+       --silent \
+       -d @<(cat <<EOF
+{
+  "ksql": "$ksqlCmd",
+  "streamsProperties": {$properties}
+}
+EOF
+))
+  printf "\n$response\n"
+  if [[ ! "$response" =~ "SUCCESS" ]]; then
+		if [[ ! "$response" =~ "already exists" ]]; then
+    	echo -e "\nERROR: KSQL command '$ksqlCmd' did not include \"SUCCESS\" in the response.  Please troubleshoot.\n"
+    	exit 1
+		fi
+  fi
+done <statements-ccloud.sql
+
+printf "\n\n====== Configuring Kibana Dashboard\n"
+./dashboard/configure_kibana_dashboard.sh
+
+printf "\n\n====== Reading data from topics and ksqlDB\n"
+CONFIG_FILE=/opt/docker/$CONFIG_FILE ./read-topics-ccloud.sh
+ 
```

Comment:
Because this demo spins up real CCloud resources that incur cost, the end of this script should include a note on how to stop the demo and destroy all CCloud resources (please see other CCloud demos for examples)

Reply:
added